dataactcore/scripts/load_historical_certified_flex_field.py

```diff
@@ -0,0 +1,184 @@
+
+    # Filling in all the shared data for these flex fields
+    now = datetime.datetime.now()
+    new_df['created_at'] = now
+    new_df['updated_at'] = now
+    new_df['job_id'] = job_id
+    new_df['submission_id'] = submission_id
+    new_df['file_type_id'] = file_type_id
+
+    return new_df
+
+
+def load_updated_flex_fields():
+    """ Load in flex fields from updated submissions as they were at the latest certification """
+    logger.info('Moving updated flex fields')
+    sess = GlobalDB.db().session
+
+    # Get a list of all submissions with certified flex fields
+    certified_flex_subs = sess.query(CertifiedFlexField.submission_id).distinct().all()
+
+    # We only want to go through updated submissions without flex fields already loaded
+    updated_subs = sess.query(Submission.submission_id).\
+        filter(~Submission.submission_id.in_(certified_flex_subs),
+               Submission.d2_submission.is_(False),
+               Submission.publish_status_id == PUBLISH_STATUS_DICT['updated']).all()
+
+    certified_ids = sess. \
+        query(func.max(CertifiedFilesHistory.certify_history_id).label('max_cert_id')). \
+        filter(CertifiedFilesHistory.submission_id.in_(updated_subs)). \
+        group_by(CertifiedFilesHistory.submission_id).cte('certified_ids')
+
+    historical_files = sess.query(CertifiedFilesHistory.filename, CertifiedFilesHistory.file_type_id,
+                                  CertifiedFilesHistory.submission_id). \
+        join(certified_ids, certified_ids.c.max_cert_id == CertifiedFilesHistory.certify_history_id).\
+        filter(CertifiedFilesHistory.file_type_id.in_(FILE_LIST))
+
+    # Loop through each updated submission
+    for historical_file in historical_files:
+        filename = historical_file.filename
+        submission_id = historical_file.submission_id
+        file_type_id = historical_file.file_type_id
+
+        # If this is a file in S3, download to a local temp file first then use temp file as local file
+        if CONFIG_BROKER['use_aws']:
+            (file, tmp_filename) = tempfile.mkstemp()
+            s3 = boto3.client('s3', region_name=CONFIG_BROKER['aws_region'])
+            s3.download_file(CONFIG_BROKER['certified_bucket'], filename, tmp_filename)
+            filename = tmp_filename
+
+        with open(filename) as file:
+            # Get file delimiter, get an array of the header row, and reset reader to start of file
+            header_line = file.readline()
+            delim = '|' if header_line.count('|') != 0 else ','
+            header_row = next(csv.reader([header_line], quotechar='"', dialect='excel', delimiter=delim))
+            file.seek(0)
+
+            flex_list = [header.lower() for header in header_row if header.lower().startswith('flex_')]
+
+            # If there are no flex fields, just ignore this file, no need to go through it
+            if len(flex_list) == 0:
+                continue
```

Comment:
Really curious how this works. Breaking/continuing in a `with` sounds questionable cause the file is still "open" as you leave but if you've tested it and it works I think it's fine.

Reply:
Technically, if you use a `with` you don't have to close the file, it closes by itself. I didn't have any problems while running my local tests